File: Data Structures & Algorithms/two-integer-sum/submission-17.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> seen;
        std::vector<int> res;

        for (size_t i = 0; i < nums.size(); i++)
        {
            int need = target - nums[i];
            if (seen.contains(need))
            {
                res.push_back(seen[need]);
                res.push_back(i);
                return res;
            }
            else
            {
                seen[nums[i]] = i;
            }
        }
        
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-17.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {} # key:num val:idx
        
        for i, n in enumerate(nums):
            need = target - n
            if need in seen:
                return [seen[need], i]
            seen[n] = i
        